servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/CompletableProcessor.java

```diff
@@ -53,9 +54,8 @@ protected void handleSubscribe(Subscriber subscriber) {
         subscriber.onSubscribe(delayedCancellable);
         if (subscribers.offer(subscriber)) {
             TerminalNotification terminalSignal = this.terminalSignal;
-            if (terminalSignal != null) {
-                // To ensure subscribers are notified in order we go through the queue to notify subscribers.
-                notifyListeners(terminalSignal);
+            if (terminalSignal != null && subscribers.remove(subscriber)) {
+                terminalSignal.terminate(subscriber);
```

Comment:
should we have a `TerminalNotification#terminate(..)` variant which doesn't throw? Otherwise the `Completable#handleSubscribe` would catch the exception and propagate a duplicate terminal event.